types/sparqljs/index.d.ts

```diff
@@ -261,19 +286,19 @@ export interface Triple {
 
 export interface PropertyPath {
     type: "path";
-    pathType: "|" | "/" | "^" | "+" | "*" | "!";
+    pathType: "|" | "/" | "^" | "+" | "*" | "!" | "?";
     items: Array<IriTerm | PropertyPath>;
 }
 
 export type Expression =
     | OperationExpression
     | FunctionCallExpression
     | AggregateExpression
-    | BgpPattern
-    | GraphPattern
-    | GroupPattern
-    | Tuple
-    | Term;
+    | Pattern // all patterns can be inside an exists.
+    | Tuple // used in IN operator
+    | IriTerm
```

Comment:
I think it is incorrect to replace `Term` here by `IriTerm | VariableTerm | LIteralTerm` because e.g. QuadTerm should be allowed here:
```sparql
BIND(<< <urn:s> <urn:p> <urn:o> >> AS ?q)
```

Reply:
Ah, good point!